### PR TITLE
Allow node draining test

### PR DESCRIPTION
**Pull request by ramperher**
- If `trex_continuous_mode` is set up, even though `trex_duration` is not set to -1, do not check job status and leave it running for future validations.
- Moving some variables to defaults (trex_app_run_passed), removing some default clauses that are not needed since the vars already have def values (e.g. enable_lb).
- Adding code to handle node draining testing.

### CODE REVIEW

**Comment by capolrik** [comment]
Very interesting!
One tricky part with the continuous burst mode of T-REX is, with the current implementation, there is no easy way to end the burst properly. And finishing the T-REX burst in a gentle way is needed to gather the statistics generated at the end.
I was trying few months ago to end it via the container itself but I faced a lot of security constraint. But I know that with a t-rex cmd how to end it as attended.
Do not hesitate to ping me on this subject :)

**Comment by ramperher** [comment]
> Very interesting! One tricky part with the continuous burst mode of T-REX is, with the current implementation, there is no easy way to end the burst properly. And finishing the T-REX burst in a gentle way is needed to gather the statistics generated at the end. I was trying few months ago to end it via the container itself but I faced a lot of security constraint. But I know that with a t-rex cmd how to end it as attended. Do not hesitate to ping me on this subject :)

Hi @capolrik ! Yes, that's why I finally decided to run a job with a limited duration, but high enough to run the node draining, reallocate the pod and wait till the end of the TRex job. At least for the moment, if then we find that we can run it and endless trex job and finish it properly, then that's another alternative.

**Comment by ramperher** [comment]
I've tested it locally and it's working fine, the draining process is quite smooth and just takes few seconds. Tomorrow I'll try with pipelines and together with trex execution

**Comment by ramperher** [comment]
This is ready for review, please check https://github.com/dci-labs/dallas-pipelines/pull/1086#issuecomment-2066369141 with my comments regarding the jobs I've launched

**Review by manurodriguez — APPROVED** [review]

File: roles/example-cnf-draining/tasks/node-draining.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  until: '"SchedulingDisabled" in nodes.stdout'
+  retries: 6
+  delay: 10
+
+# Not really needing a complete draining, just removing testpmd-app pod
+# Running this in asynchronous mode, then we will wait until pod is deleted
+- name: Drain the node to remove testpmd-app pod
+  ansible.builtin.shell: |
+    {{ oc_tool_path }} adm drain {{ cnf_existing_node }} --pod-selector example-cnf-type=cnf-app --delete-emptydir-data
+  async: 60
```
[the remark this answers is not in the document]
Comment (ramperher) [reply roles/example-cnf-draining/tasks/node-draining.yml:22]: Yes, we could do a full draining, I tested it and it takes of course more time. But at the end, what we want to do is to move the testpmd-app pod from one worker to another, so I think, for the moment, it is good to just drain that pod.

Regarding the results, yes, it's interesting, and what's more interesting is that, in the first test I launched with 30 minutes, packet loss was 0 and the test passed. So I don't know if, by increasing the trex duration, we could manage to make the job to pass somehow. That's why I want to test this on dailies, to have multiple jobs to check and to confirm if there's a kind of trend here.

**Pull request merged**